src/client_side.cc

```diff
@@ -657,28 +656,26 @@ clientSetKeepaliveFlag(ClientHttpRequest * http)
 }
 
 /// checks body length of non-chunked requests
-static int
+static bool
 clientIsContentLengthValid(HttpRequest * r)
 {
     // No Content-Length means this request just has no body, but conflicting
     // Content-Lengths mean a message framing error (RFC 7230 Section 3.3.3 #4).
     if (r->header.conflictingContentLength())
-        return 0;
-
-    switch (r->method.id()) {
-
-    case Http::METHOD_GET:
+        return false;
 
-    case Http::METHOD_HEAD:
-        /* We do not want to see a request entity on GET/HEAD requests */
-        return (r->content_length <= 0 || Config.onoff.request_entities);
+    // RFC 7230 section 3.3 - Content-Length defines body exists regardless of method
+    if (r->content_length > 0)
+        return true;
 
-    default:
-        /* For other types of requests we don't care */
-        return 1;
-    }
+    // We do not want a request entity on HTTP/1.0 GET/HEAD requests
+    // without a Content-Length header.
+    const auto m = r->method.id();
+    if (m == Http::METHOD_GET || m == Http::METHOD_HEAD)
+        return !(r->http_ver <= Http::ProtocolVersion(1,0));
 
-    /* NOT REACHED */
+    // For other types of requests we don't care
```

Comment:
It is not clear to me what "do not care" means in this context. We are probably returning `true` for a specific reason. I suspect that reason is RFC requirements/rules. The answer may depend on the function name/purpose which might change during this PR.

Reply:
I believe it means the caching issues that are mentioned do not exist for other methods. Likely because they are not cacheable.

Reply:
comment removed in fc31d38